### PR TITLE
Fix `Query condition missed key schema element` error in DynamoDB

### PR DESCRIPTION
**Pull request by brfrn169**
`Query condition missed key schema element` error happens when executing Scan/Get operations using secondary index and specifying projections in DynamoDB. This PR fixes this bug. Please take a look!

### CODE REVIEW

File: core/src/test/java/com/scalar/db/storage/dynamo/SelectStatementHandlerTest.java [hunk 1]
```diff
@@ -225,6 +225,42 @@ public void handle_ScanOperationWithIndexGiven_ShouldCallQuery() {
     assertThat(actualRequest.expressionAttributeValues()).isEqualTo(expectedBindMap);
   }
 
+  @Test
+  public void handle_ScanOperationWithIndexGiven_WithProjections_ShouldCallQuery() {
```
Comment (brfrn169) [inline core/src/test/java/com/scalar/db/storage/dynamo/SelectStatementHandlerTest.java:229]: This is a unit test to reproduce the bug.

File: core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java [hunk 2]
```diff
@@ -109,15 +110,17 @@ private Scanner executeQueryWithIndex(Selection selection, TableMetadata tableMe
     ValueBinder binder = new ValueBinder(DynamoOperation.VALUE_ALIAS);
     keyColumn.accept(binder);
     Map<String, AttributeValue> bindMap = binder.build();
-    builder
-        .keyConditionExpression(condition)
-        .expressionAttributeValues(bindMap)
-        .expressionAttributeNames(ImmutableMap.of(expressionColumnName, column));
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java:115]: This bug is introduced in #462.

We set `expressionAttributeNames` here, but we also set it in `projectionExpression()` if the operation has projections, so the `expressionAttributeNames` set here is overwritten, which is a problem.
https://github.com/scalar-labs/scalardb/pull/544/files#diff-59d53393b90fb6d234264b0143965e0670ab3a4e80341de6a4e79b3bf1d3c29aL177-R188

File: core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java [hunk 3]
```diff
@@ -109,15 +110,17 @@ private Scanner executeQueryWithIndex(Selection selection, TableMetadata tableMe
...
+
+    Map<String, String> expressionAttributeNames = new HashMap<>();
+    expressionAttributeNames.put(expressionColumnName, column);
 
     if (!selection.getProjections().isEmpty()) {
-      projectionExpression(builder, selection);
+      projectionExpression(builder, selection, expressionAttributeNames);
     }
 
+    builder.expressionAttributeNames(expressionAttributeNames);
```
Comment (brfrn169) [inline core/src/main/java/com/scalar/db/storage/dynamo/SelectStatementHandler.java:122]: In this fix, we first merge the `expressionAttributeNames` and set it to the builder.

**Review by feeblefakie — DISMISSED** [review]
LGTM! Thank you!

**Review by ypeckstadt — APPROVED** [review]
LGTM. Thank you.

**Review by thongdk8 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**